File: Modix/ClientApp/src/models/infractions/InfractionCreationData.ts
import { InfractionType } from './InfractionType';

export default class InfractionCreationData
{
    type: InfractionType = InfractionType.Notice;
    reason: string = "";
    durationMonths: number | null = null;
    durationDays: number | null = null;
    durationHours: number | null = null;
    durationMinutes: number | null = null;
    durationSeconds: number | null = null;
}

File: Modix/Models/InfractionCreationData.cs
namespace Modix.Models
{
    public class InfractionCreationData
    {
        public string Type { get; set; }

        public string Reason { get; set; }

        public int? DurationMonths { get; set; }

        public int? DurationDays { get; set; }

        public int? DurationHours { get; set; }

        public int? DurationMinutes { get; set; }

        public int? DurationSeconds { get; set; }
    }
}
